opencog/atoms/bind/PatternLink.cc

```diff
@@ -506,14 +548,8 @@ void PatternLink::unbundle_virtual(const std::set<Handle>& vars,
 		{
 			_pat.evaluatable_terms.insert(sh);
 			add_to_map(_pat.in_evaluatable, sh, sh);
-			// But they're virtual only if they have two or more
-			// unquoted, bound variables in them. Otherwise, they
-			// can be evaluated on the spot.
```

Comment:
Doing it this way (ie. not evaluate on the spot) also allow the same atom `x < y` to appear in multiple clauses, so that in one clause it is virtual and should be evaluated, and in the other it should be matched as is.  This is test in the new unit test.